Implement worker script to process inventory and availability update messages

christine-w opened the pull request.

christine-w committed:
Support queue specification when polling for new messages

christine-w committed:
Add queue specification for message deletion

christine-w committed:
Implement db helper functions to update listings and availability tables

christine-w committed:
Update analytics worker to specify queue when checking messages

christine-w committed:
Implement worker script to process inventory and availability update messages

tiffaphm commented on messageBus/amazonSQS.js:
```diff
@@ -41,12 +41,12 @@ module.exports.poll = (queue) => {
 };
 
-module.exports.done = (messages) => {
+module.exports.done = (queue, messages) => {
```
Just a point of clarification - is this so that you can use the function for different queues?

christine-w replied:
Yes - this is so that I can use the same methods regardless of the queue.

tiffaphm reviewed (approved):
Code looks more modular now.

christine-w committed:
Update availability listing-date index to be unique

christine-w committed:
Handle bad updates from Inventory (invalid listings)
Current version simply discards the message. Future versions should trigger alerts that can be picked up by overall system monitoring.
